elide-integration-tests/src/test/java/com/yahoo/elide/tests/ResourceIT.java

```diff
@@ -443,72 +443,61 @@ public void testReadPermissionWithFilterCheckCollectionId() {
         // filterExpressionCheckObj.id != User'id.
 
         //test root object collection, should just receive 2 out of 3 records.
-        String getResult1 = given()
+        String expected1 = "{\"data\":[{\"type\":\"filterExpressionCheckObj\",\"id\":\"1\",\"attributes\":{\"name\":\"obj1\"},\"relationships\":{\"listOfAnotherObjs\":{\"data\":[{\"type\":\"anotherFilterExpressionCheckObj\",\"id\":\"1\"}]}}},{\"type\":\"filterExpressionCheckObj\",\"id\":\"2\",\"relationships\":{\"listOfAnotherObjs\":{\"data\":[]}}}]}";
+        given()
                 .contentType(JSONAPI_CONTENT_TYPE)
                 .accept(JSONAPI_CONTENT_TYPE)
                 .get("/filterExpressionCheckObj")
                 .then()
                 .statusCode(HttpStatus.SC_OK)
-                .extract().response().asString();
+                .body(equalTo(expected1));
 
         //test authentication pass querying with ID == 1
-        String getResult2 = given()
+        String expected2 = "{\"data\":{\"type\":\"filterExpressionCheckObj\",\"id\":\"1\",\"attributes\":{\"name\":\"obj1\"},\"relationships\":{\"listOfAnotherObjs\":{\"data\":[{\"type\":\"anotherFilterExpressionCheckObj\",\"id\":\"1\"}]}}}}";
+        given()
                 .contentType(JSONAPI_CONTENT_TYPE)
                 .accept(JSONAPI_CONTENT_TYPE)
                 .get("/filterExpressionCheckObj/1")
                 .then()
                 .statusCode(HttpStatus.SC_OK)
-                .extract().response().asString();
+                .body(equalTo(expected2));
 
         //test authentication pass querying with ID == 2, it shouldn't contain attribute "name".
-        String getResult3 = given()
+        String expected3 = "{\"data\":{\"type\":\"filterExpressionCheckObj\",\"id\":\"2\",\"relationships\":{\"listOfAnotherObjs\":{\"data\":[]}}}}";
+        given()
                 .contentType(JSONAPI_CONTENT_TYPE)
                 .accept(JSONAPI_CONTENT_TYPE)
                 .get("/filterExpressionCheckObj/2")
                 .then()
                 .statusCode(HttpStatus.SC_OK)
-                .extract().response().asString();
+                .body(equalTo(expected3));
 
         //test authentication fail querying with ID == 3
-        String getResult4 = given()
+        given()
                 .contentType(JSONAPI_CONTENT_TYPE)
                 .accept(JSONAPI_CONTENT_TYPE)
                 .get("/filterExpressionCheckObj/3")
                 .then()
-                .statusCode(HttpStatus.SC_NOT_FOUND)
-                .extract().response().asString();
+                .statusCode(HttpStatus.SC_NOT_FOUND);
```

Comment:
compare to expected4?

Reply:
No, there was no expected4.  We could add one?
```java
        assertEquals(getResult3, expected3);
        assertEquals(getResult5, expected5);
        assertEquals(getResult6, expected5);
```

Reply:
It would resemble
```json
{
    "errors": [
        {
            "detail": "Unknown identifier 3 for filterExpressionCheckObj"
        }
    ]
}
```